tasty-query/shared/src/main/scala/tastyquery/Symbols.scala

```diff
@@ -417,6 +417,78 @@ object Symbols {
       */
     final def isStableMember(using Context): Boolean =
       !isAnyOf(Method | Mutable) && !declaredType.isInstanceOf[ExprType]
+
+    /** The symbol actually implementing this symbol for an object of class `runtimeClass`.
+      *
+      * If this symbol is private or if it is a constructor,
+      * `runtimeImplementingSymbol` returns this symbol.
+      *
+      * Otherwise, it returns the first non-abstract symbol matching this
+      * symbol in the linearization of `runtimeClass`.
+      *
+      * @throws java.lang.IllegalArgumentException
+      *   if `this.owner.isClass` is false or if
+      *   `runtimeClass.isSubclass(this.owner.asClass)` is false
+      * @throws tastyquery.Exceptions.MemberNotFoundException
+      *   if no concrete matching symbol could be found
+      */
+    final def runtimeImplementingSymbol(runtimeClass: ClassSymbol)(using Context): TermSymbol =
+      if canMatchInheritedSymbols then
+        val linearization = runtimeClass.linearization
+        if !linearization.contains(this.owner.asClass) then
+          throw IllegalArgumentException(s"Invalid run-time class $runtimeClass for dynamic super call to $this")
+        findRuntimeSymbolCommon(runtimeClass, linearization)
+      else
+        if !this.owner.isClass then
+          throw IllegalArgumentException(s"Nonsensical run-time implementation lookup for local symbol $this")
+        this
+    end runtimeImplementingSymbol
+
+    /** Resolves a dynamically bound `super` call for an object of class `runtimeClass`.
+      *
+      * This method returns the next symbol for a `super` call of the form
+      * `enclosingClass.super.foo` for an object of class `runtimeClass`.
+      * That is the first non-abstract symbol matching this symbol in the
+      * linearization of `runtimeClass`, after the occurrence of
+      * `enclosingClass`.
+      *
+      * @throws java.lang.IllegalArgumentException
+      *   if any of the following conditions happen:
+      *   `this.owner.isClass` is false;
+      *   or this symbol is private;
+      *   or this symbol is a constructor;
+      *   or `runtimeClass.isSubclass(this.owner.asClass)` is false;
+      *   or if `runtimeClass.isSubclass(enclosingClass)` is false
+      * @throws tastyquery.Exceptions.MemberNotFoundException
+      *   if no concrete matching symbol could be found
+      */
+    final def nextSuperSymbol(runtimeClass: ClassSymbol, enclosingClass: ClassSymbol)(using Context): TermSymbol =
+      if !canMatchInheritedSymbols then
+        throw IllegalArgumentException(s"Invalid dynamic super call to statically bound symbol $this")
```

Comment:
How about indicating that the symbol is private or a constructor? I am not sure I would understand what "statically bound" means if I see this error message :‌D